files/config/kube-proxy.yaml

```
@@ -3,7 +3,7 @@ clientConnection:
   kubeconfig: /etc/kubernetes/config/proxy-kubeconfig.yaml
 kind: KubeProxyConfiguration
 mode: iptables
-{{- if and (ne .Cluster.Kubernetes.CloudProvider "aws") (not .DisableCalico) }}
+{{- if and (and (ne .Cluster.Kubernetes.CloudProvider "aws") (ne .Cluster.Kubernetes.CloudProvider "azure")) (not .DisableCalico) }}
```

Won't that be the same? Do we disable calico on kvm?

```suggestion
{{- if eq .Cluster.Kubernetes.CloudProvider "kvm" }}
```

---

there might be a problem with AWS legacy release where we would use this k8scc as on legacy AWS we still sue calico for CNI

---

Yes, precisely now, but it would be annoying to hunt down when someone adds another provider. I'd rather explicitly exclude the specific known ones and let others decide.

I did have problems parsing that logic in my head as well, but I think it's smaller issue than having to hunt weird problems down on new provider.

---

👍